src/formatDate.js

```diff
@@ -9,6 +9,34 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const separator = date.split(fromFormat[3]);
```

Comment:
Using ``fromFormat[3]`` to define the delimiter is not flexible because it assumes that the delimiter is always in the 4th position, which may not always be correct. The delimiter should be defined as the last element of the format array.

Example:

```suggestion

  // Change: Define delimiter as the last fromFormat element
  const fromSeparator = fromFormat[fromFormat.length - 1];
  const toSeparator = toFormat[toFormat.length - 1];
  
  // Change: split date by separator
  const dateParts = date.split(fromSeparator);
```

Edit the code for these changes.